### PR TITLE
[checklists] Remove reviewer/signoff lines for D/V 1-2 and align checklist discrepancies

### PR DESCRIPTION
**Pull request by msfschaffner**
This PR removes the signoff/reviewer items from the D1, D2, V1, V2 milestones? @sjgitty is this along the lines of what you intended?

I also went over the existing checklists and aligned them with the newest version of the template. We've accumulated quite a few discrepancies in the mean time. And it turned out that all checklist items links where broken :D.

If this is OK, I'll put it up for wider review such that the IP owners can go through and double check.

### CODE REVIEW

**Comment by msfschaffner** [comment]
CC @lenary

**Comment by imphil** [comment]
Talking to @lenary before, we can remove also the reviews from the software checklist except for the S3 state to match what HW is doing.

**Comment by sjgitty** [comment]
> is this along the lines of what you intended?

Yes-ish. It is huge, which I imagine is due to the discrepancies that you are referring to. Have you done your own "smart diff" to make sure that the only things that are changing are `-` to `_`, etc? I could do a cross-check if that helps.

**Comment by msfschaffner** [comment]
RE smart-diff: yes and no. I had to resolve several discrepancies by hand. 

This is why I've split the changes into 3 commits. The first one just removes the "reviewers" and "signed off" rows, and the last one just replaces `-` by `_` (which was essentially a search replace op in multiple files). 

The commit that involved some manual work is the middle one, and it would be great if you could double check that this is indeed correct.

File: hw/ip/rv_core_ibex/doc/checklist.md
Comment (msfschaffner) [inline hw/ip/rv_core_ibex/doc/checklist.md:122]: Ibex is already in V1. Aligning this to the newest template adds a few categories like `CSR_CHECK_GEN_AUTOMATED`, `FPV_MAIN_ASSERTIONS_PROVEN` and `FPV_REGRESSION_SETUP`. 
We can probably set these three to `N/A` - since Ibex has no CSRs and no FPV testbench - but I wanted to double check.

Comment (sjgitty) [reply hw/ip/rv_core_ibex/doc/checklist.md:122]: I agree, but would be good to get confirmation from @udinator and @rswarbrick

File: hw/ip/gpio/doc/checklist.md
Comment (sjgitty) [inline hw/ip/gpio/doc/checklist.md:37]: any reason why the d1-reviewed (and d2- and d3-) are still using dash vs underscore?

**Review by sjgitty — APPROVED** [review]
I certainly didn't check every single line :) but I spot checked all three commits. Left two comments. Thanks for doing this @msfschaffner it is a labor of love :). Perhaps we need a linter for our signoff files :D (please don't)

File: hw/ip/rv_core_ibex/doc/checklist.md
Comment (sriyerg) [reply hw/ip/rv_core_ibex/doc/checklist.md:122]: Ibex is verified in a separate repo. We verify this wrapper at the chip level (i.e. close the coverage on this module). Our [dashboard](https://docs.opentitan.org/hw/#processor-cores) has a separate heading for Ibex, which is where the status (and Ibex's checklist) should be tracked. IMO having a checklist for this testbench-less module does not seem useful.

Comment (sriyerg) [reply hw/ip/rv_core_ibex/doc/checklist.md:122]: Alternately, we treat this the same way as `rstmgr` and others verified at chip level.

File: hw/ip/gpio/doc/checklist.md
Comment (msfschaffner) [reply hw/ip/gpio/doc/checklist.md:37]: Ah these have been removed in all other files and the template, except for these two instances. That's why they still used the old format. I've deleted them from this checklist as well.

**Comment by msfschaffner** [comment]
@lenary I've updated the SW checklists as well. can you PTAL? I'll squash the commits once approved.

**Review by lenary — APPROVED** [review]
SW Changes LGTM

File: hw/ip/rv_core_ibex/doc/checklist.md
Comment (msfschaffner) [reply hw/ip/rv_core_ibex/doc/checklist.md]: Shall we move on and merge this, and decide later on whether to remove the Ibex checklist or not?

Comment (imphil) [reply hw/ip/rv_core_ibex/doc/checklist.md]: Yeah, please file a follow-up to continue the discussion there.

**Review by imphil — APPROVED** [review]
LGTM (both in approach and for the OTBN changes)

File: hw/ip/rv_core_ibex/doc/checklist.md
Comment (msfschaffner) [reply hw/ip/rv_core_ibex/doc/checklist.md]: Tracked here: #3808

**Comment by msfschaffner** [comment]
Thanks for the reviews, merging!

**Pull request merged**
